api_server.py

```
@@ -159,9 +163,11 @@ def check_rules(available_ids: List[int]) -> Dict:
             "ba_non_ttr": 0,
         }
     with get_db() as conn:
+        # Construct placeholders based on number of IDs (Safe with parameterization)
         placeholders = ",".join("?" * len(available_ids))
         rows = conn.execute(
-            f"SELECT role, skills FROM crew WHERE id IN ({placeholders})", available_ids
+            f"SELECT role, skills FROM crew WHERE id IN ({placeholders})",  # nosec B608
```

Ruff (0.15.7)

[error] Possible SQL injection vector through string-based query construction

(S608)